src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs

```diff
@@ -228,6 +228,29 @@ public virtual IEnumerable<string> AreaViewLocationFormats
             return new RazorPageResult(pageName, searchedLocations);
         }
 
+        private string GetNormalizedRouteValue(string key, ActionContext context)
+        {
+            if (context.ActionDescriptor.AttributeRouteInfo == null)
+            {
+                var match = context.ActionDescriptor.RouteConstraints.SingleOrDefault(r => r.RouteKey == key);
```

Comment:
The additional thing you have to check with RouteConstraints is the keytype. You only want to normalize if it's a requiredmatch (or whatever). If it's a deny then the value must be null, If it's a catch-all you need to use the route value.